tests/end-to-end/upgrade/main.go

```diff
@@ -205,6 +210,7 @@ func getDexConfigFromCluster(k8sCli *k8sClientSet.Clientset, userSecret, dexName
 		secret, err := k8sCli.CoreV1().Secrets(dexNamespace).Get(userSecret, metav1.GetOptions{})
 		if err != nil {
 			logrus.Infof("while getting dex secret: %v", err)
+			time.Sleep(5 * time.Second)
```

Comment:
I'd like to remove this sleep  or put a comment why in this place you really need 5 seconds. On the other hand the waiter util has 1000 milliseconds sleep so maybe it is enough